Code/Framework/AzNetworking/AzNetworking/Serialization/NetworkInputSerializer.cpp

```diff
@@ -128,20 +123,20 @@ namespace AzNetworking
     {
         m_serializerValid &= (inputValue >= minValue);
         m_serializerValid &= (inputValue <= maxValue);
-        const uint64_t valueRange = static_cast<uint64_t>(maxValue - minValue);
+        const uint64_t valueRange = static_cast<uint64_t>(maxValue) - static_cast<uint64_t>(minValue);
         if (valueRange <= AZStd::numeric_limits<uint8_t>::max())
         {
-            return SerializeBoundedValueHelper<uint8_t>(static_cast<uint8_t>(inputValue - minValue));
+            return SerializeBoundedValueHelper<uint8_t>(static_cast<uint8_t>(inputValue) - static_cast<uint8_t>(minValue));
```

Comment:
I don't think you wanted to cast this part like this..? In order to minimize risk of underflow/overflow, it's better to do all the calculations in the largest register space available and only cast the final result after subtraction.

Reply:
As a case in point, consider a uint16_t type where minValue was 500 and maxValue was 510 or something.. there's only 10 integers in that range, so it can easily be transmitted in a byte or less even. 510 - 500 == 10, easily castable to a uint8_t without any risk of loss of value, however casting 500 and 510 individually to uint8_t's absolutely will truncate and lose data.